### ISSUE TITLE
Document why celery is gross and how this lib overcomes that flaws 

### ISSUE THREAD

**Issue by Ambro17** [closeio/tasktiger#141]
I never used the lib and I would like to read in a special section a comparison of approaches and advantages of each other.
Also the motivation to write a task queue from the ground up would be nice

**Comment by pavank** [comment]
I am curious about this too.  I use celery in my projects, and while the ever-changing API drives me mad at times, the vast amount of community resources available outweigh this annoyance.  I would like to know what design aspects make this library worth spending my time to learn at the cost of ditching Celery.

**Comment by AlecRosenbaum** [comment]
The biggest design aspect of this library that addresses issues I've had with `celery` in the past is that `tasktiger` relies specifically on Redis and does not support a generalized broker. Depending on how you look at it, this could be either a feature or a flaw. In my opinion, this is a positive point. Not only does it reduce complexity in terms of testing/validating functionality, but it also allows `tasktiger` to take advantage of some of Redis's built-in data structures that simplify entire classes of features. For example: using [`zset`s simplifies delayed tasks](https://redislabs.com/ebook/part-2-core-concepts/chapter-6-application-components-in-redis/6-4-task-queues/6-4-2-delayed-tasks/) immensely compared to `celery`'s paradigm of workers consuming delayed tasks and holding on to them until they're ready to execute.

Other smaller things include:
* Tasks execute in a forked worker, which prevents memory-leak issues caused by underlying C implementations of python libraries (like `imagemagick`, etc.)
* Workers are designed to gracefully work on multiple queues/subqueues, easily allowing for things like "Flexible queues" (e.g. create subqueues for each customer to prevent one customer from hogging all the compute capacity). I'm not sure that celery has the concept of subqueues.
* Tasks' arguments _must_ be JSON serializable. In recent versions of celery the JSON serializer is the default, but pickling objects is still supported and used pretty widely. Providing pickled objects to tasks not only can be confusing when a task gets an outdated model instance as an argument (for example), but it also results in storing a _lot_ more data in the broker.
* In my experience, Flower never worked well enough as a monitoring solution to be useful and celery was too complicated to build anything useful on my own. In contrast, I've found `tasktiger-admin` to be quite helpful on its own and easy enough to customize that I can pretty easily add an admin page into my application for more custom monitoring.
* Unique tasks work fabulously well and eliminate a major pain point I had with duplicate celery tasks.
* Scheduled/periodic tasks are more graceful - I don't have to make sure I have a different `celery beat` worker running to get tasks to run on a regular basis. This builds off of unique tasks in that effectively all workers just run with celery's `-B` option, but since tasks are unique it doesn't matter if workers try to schedule duplicates.

Just in general, I've had a much better experience going through `tasktiger`'s sourcecode. I've lost days to trying to debug issues in celery, largely because there's multiple complex repos that interact with each other in non-obvious ways. By contrast, I've been able to debug, fix, and test similarly sized `tasktiger` issues within a few hours.

I don't know that this makes sense to put into the `README`, but I hope it's useful for you.
